toolchain/parser/parser_handle_period.cpp

```diff
@@ -39,15 +42,23 @@ auto ParserHandlePeriod(ParserContext& context, ParseNodeKind node_kind)
 }
 
 auto ParserHandlePeriodAsDeclaration(ParserContext& context) -> void {
-  ParserHandlePeriod(context, ParseNodeKind::QualifiedDeclaration);
+  ParserHandlePeriodOrArrow(context, ParseNodeKind::QualifiedDeclaration,
+                            false);
```

Comment:
```suggestion
                            /*is_arrow=*/false);
```

Maybe `/*is_arrow=*/`? I know people can look up to the function, but slight preference for marking bools just as a habit.